fe1-web/src/features/social/components/__tests__/NewChirp.test.tsx

```diff
@@ -81,9 +82,51 @@ describe('NewChirp', () => {
     fireEvent.changeText(getByTestId('new_chirp_input'), mockText);
     fireEvent.press(getByTestId('new_chirp_publish'));
 
-    await waitFor(() => {
-      expect(requestAddChirp).toHaveBeenCalledWith(mockPopToken.publicKey, mockText, mockLaoId);
-      expect(requestAddChirp).toHaveBeenCalledTimes(1);
-    });
+    expect(requestAddChirp).toHaveBeenCalledWith(mockPopToken.publicKey, mockText, mockLaoId);
+    expect(requestAddChirp).toHaveBeenCalledTimes(1);
+  });
+
+  it('shows the modal on trimmed chirp and closes it', async () => {
+    const { toJSON, getByTestId, findByText } = render(
+      <Provider store={mockStore}>
+        <FeatureContext.Provider value={contextValue}>
+          <SocialMediaContext.Provider value={socialContextValue}>
+            <NewChirp />
+          </SocialMediaContext.Provider>
+        </FeatureContext.Provider>
+      </Provider>,
+    );
+    const mockText = '\nThis    text\n \twill be trimmed \n ';
+
+    // Add an invalid chirp
+    fireEvent.changeText(getByTestId('new_chirp_input'), mockText);
+    fireEvent.press(getByTestId('new_chirp_publish'));
+
+    expect(getByTestId('confirm-modal-confirm')).toBeDefined();
+    expect(toJSON()).toMatchSnapshot();
+
+    // Accept the modal message
+    fireEvent.press(getByTestId('confirm-modal-confirm'));
+
+    expect(await findByText('300')).not.toBeNull();
```

Comment:
The `300` is not ideal but I don't have a better idea right this second sooo 👍

Reply:
Tried a new one, but not necessary. better